lib/nerves_hub_link/device_channel.ex

```diff
@@ -71,8 +71,15 @@ defmodule NervesHubLink.DeviceChannel do
   end
 
   def handle_info(%Message{event: "update", payload: update}, state) do
-    _ = UpdateManager.apply_update(update)
-    {:noreply, state}
+    case NervesHubLinkCommon.Message.UpdateInfo.parse(update) do
+      {:ok, %NervesHubLinkCommon.Message.UpdateInfo{} = info} ->
```

Comment:
It really feels like we should be able to trust that if `UpdateInfo.parse/1` returns `:ok`, that it does return an `UpdateInfo` struct. Are you worried about it not doing that?

Reply:
no, i just always put the struct name so the reader doesn't have to go look at the docs of that function.